### PR TITLE
Fixes to the feed discovery page

### LINKED ISSUE #1204: feed search results are cleared when you open a feed and go back
**Describe the bug**

When you go to the feed search ("Discover Feeds") and type something, you get a list of results. However, that list reverts back to the default suggestions when you look into one of the results and go back to the list.

**To Reproduce**

1. Open "My Feeds"
2. Click the "Discover new feeds" search box
3. Type e.g. "linux" -> a list of results matching linux appears
4. Click a random entry on the results list
5. Click the back arrow
6. The list very briefly shows the previous results, and after less than a second the search results are replaced with the default feed suggestions, the ones that are shown before you start typing. The entered query "linux" is still visible in the search field. To get the results again, you need to focus the search field and press enter.

**Expected behavior**

The results should not be cleared until you press the X in the search field or "Cancel".

**Screenshots**

<p></p>

**Details**

 - Platform: iOS, web

### LINKED ISSUE #1236: [APP-832] Remove extraneous "loading" element in feed discovery
There's both a loading spinner and a "Loading..." element in the feed discovery view. It seems like it might look right on web but not mobile. Clean that up.

<sub>[APP-832](https://linear.app/blueskyweb/issue/APP-832/remove-extraneous-loading-element-in-feed-discovery)</sub>

### PR DESCRIPTION
**Pull request by pfrazee**
Fixes some styling of the loading and empty states (close #1236)
Maintains the search query in feed discovery across navigations (close #1204)

### CODE REVIEW

**Comment by pfrazee** [comment]
Yeesh that yarn.lock update is intense. I had to nuke my node modules after testing something random. Gonna see if anything is screwy there.

**Comment by pfrazee** [comment]
Yarn.lock update seems fine? We just must not have updated in a while.

File: tsconfig.json [hunk 1]
```diff
@@ -2,11 +2,12 @@
   "extends": "@tsconfig/react-native/tsconfig.json",
   "compilerOptions": {
     "jsx": "react-jsx",
+    "types": ["node"],
```
Comment (estrattonbailey) [inline tsconfig.json:5]: Node?

**Review by estrattonbailey — APPROVED** [review]
LGTM, just the search issue I Slacked you about

File: tsconfig.json (hunk 1 above)
Comment (pfrazee) [reply tsconfig.json:5]: Yeah, we use some node types in places (NodeJS.Timer for instance)

**Review by estrattonbailey — APPROVED** [review]
Can't repro that issue now, so LGTM!

**Pull request merged**